### PR TITLE
Add validator to perform offline CRD validation in tests

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: pkg/config/crd/validator.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+		obj, _, err := decoder.Decode([]byte(item), nil, nil)
+		if err != nil {
+			return err
+		}
+		errs = multierror.Append(errs, v.ValidateCustomResource(obj))
+	}
+	return errs.ErrorOrNil()
+}
+
+func (v *Validator) ValidateCustomResource(o runtime.Object) error {
```
Comment (jasonwzm) [inline pkg/config/crd/validator.go:69]: How confident are we with this approach?

How do we plan to use this validator with our config files? Can we setup a pipeline in presubmit that takes config files and dump into a live k8s cluster and validate there?

Comment (howardjohn) [reply pkg/config/crd/validator.go:69]: It's not 100% accurate as noted in the workaround part where we filter out some errors. I would hope that at some point we figure out how to do this properly. I do not want our unit tests requiring a live cluster.

This isn't perfect but it gives us strictly more coverage than today do I think it's still useful. I wouldn't expose to users (like integrate with analyze) until we iron out the kinks though

Comment (jasonwzm) [reply pkg/config/crd/validator.go:69]: How do we ensure coverage on config files?

Comment (howardjohn) [reply pkg/config/crd/validator.go:69]: I am not sure what you mean?

Current state: we have a bunch of configs that may be invalid

With this PR: we have a bunch of configs that may be invalid, but for some of them that we call ValidateCustomResource we have some degree of confidence they are valid

Comment (jasonwzm) [reply pkg/config/crd/validator.go:69]: So the next step is to have all "readconfig" funcs use the validator?

Comment (howardjohn) [reply pkg/config/crd/validator.go:69]: yes, if they test wants to enforce the configs passed are valid they should do this. Currently I have added it to one test where its a major issue and constantly falling out of sync with the schema validation

Comment (jasonwzm) [reply pkg/config/crd/validator.go:69]: Ok. Can we file a bug or add a TODO that we need to provide a more reliable way for validation testing? Something that we can keep track.

**Review by jasonwzm — APPROVED** [review]
LGTM based on the current situation.

**Review by linsun — APPROVED** [review]
licenses added LGTM

**Comment by linsun** [comment]
| /test integ-multicluster-k8s-tests_istio
-- | --

**Pull request merged**